### PR TITLE
Add MessageHeaderAttribute into the Contract.

### LINKED ISSUE #631: Add MessageHeaderAttribute into the Contract

### PR DESCRIPTION
**Pull request by shmao**
Fix #631.

### CODE REVIEW

**Comment by shmao** [comment]
@dotnet-bot test outerloop please

File: src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs [hunk 1]
```diff
@@ -379,6 +379,14 @@ public partial class InvalidMessageContractException : System.Exception
         public InvalidMessageContractException(string message) { }
         public InvalidMessageContractException(string message, System.Exception innerException) { }
     }
+
```
Comment (shmao) [inline src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs]: Remove this line.

**Comment by iamjasonp** [comment]
We may have to rev the version of the contract with a contract addition as well. Changes otherwise LGTM

**Comment by iamjasonp** [comment]
To clarify - if there are changes needed to rev the version number, please include them in the this PR before merging... thanks

File: src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs [hunk 2]
```diff
@@ -379,6 +379,14 @@ public partial class InvalidMessageContractException : System.Exception
...
+    [System.AttributeUsageAttribute((System.AttributeTargets)(384), AllowMultiple = false, Inherited = false)]
```
Comment (mconnew) [inline src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs]: Although we haven't done it elsewhere, I really don't like using the magic numbers and casting them to the relevant enum type. The original value was ServiceModelAttributeTargets.MessageMember which isn't public, but that equates to AttributeTargets.Property | AttributeTargets.Field.

File: src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs [hunk 3]
```diff
@@ -379,6 +379,14 @@ public partial class InvalidMessageContractException : System.Exception
...
+    public class MessageHeaderAttribute : MessageContractMemberAttribute
+    {
+        public bool MustUnderstand { get { return default(bool); } set { } } 
+        public bool Relay { get { return default(bool); } set { } }
```
Comment (mconnew) [inline src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs]: Are we doing anything with the Relay field? If not, then should we be including it?

Comment (shmao) [inline src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs]: Yes, the field is used somewhere in the code.

File: src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs (hunk 2 above)
Comment (shmao) [inline src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs]: Other APIs are already using this magic number. I think we should just make it consistent with other types.

**Comment by shmao** [comment]
Removed MessageHeaderAttribute.Relay and MessageHeaderAttribute.Actor from the contract per @mconnew 's suggestion.

@dotnet-bot test outerloop please

**Comment by shmao** [comment]
@dotnet-bot test outerloop please

**Comment by mconnew** [comment]
LGTM

**Pull request merged**